### ISSUE TITLE
Brain Dumps: PhaseTracker Edition

### ISSUE THREAD

**Issue by gabizou** [SpongePowered/SpongeDocs#705]
So, since @Inscrutable asked for some more information on topics to document for the SpongeDocs, one of the priority topics is the PhaseTracker for the implementation in SpongeCommon/SpongeForge/SpongeVanilla.

So..... Without much to do:

PhaseTracker explained in a gist with Markdown:

https://gist.github.com/gabizou/e4d0e9e7fee420612c3519e14c450f16

**Comment by Inscrutable** [comment]
I can probably start drafting that into something that looks more like our regular style, unless anyone else wants to volunteer. But so far (at a glance) it seems to consist largely of *how it works*, without much on how it can be used (or abused). It is critical to know what you're getting, so the nuts-and-bolts breakdown is useful, but there are probably many use-case needs that aren't yet satisfied. I welcome more opinions on this as I haven't any more than the foggiest idea myself.

**Comment by gabizou** [comment]
Right, I haven't quite finished explaining how it's working, but I'm about to get to how it works once I explain about PhaseContexts.

**Comment by Inscrutable** [comment]
OK, Now I get that this section would primarily be aimed toward contributing to sponge, and not exposing it as an API. So this belongs under `contributing/implementation`.
